x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/textstructure/structurefinder/TextStructure.java

```diff
@@ -403,6 +426,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         if (grokPattern != null && grokPattern.isEmpty() == false) {
             builder.field(GROK_PATTERN.getPreferredName(), grokPattern);
         }
+        if (ecsCompatibility != null && ecsCompatibility.isEmpty() == false) {
+            builder.field(ECS_COMPATIBILITY.getPreferredName(), ecsCompatibility);
+        }
```

Comment:
If you enforce it to never be `null` or empty with `getNonNullEcsCompatibilityString()` then you don't need to check here:

```suggestion
        builder.field(ECS_COMPATIBILITY.getPreferredName(), ecsCompatibility);
```